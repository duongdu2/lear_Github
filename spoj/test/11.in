71
17
762061862 924702843 564862054 811871591 430566554 306124396 447978610 286209008 265649417 995486696 649509845 765681176 82237277 78937137 989969511 605839825 146309323 
48
973793217 856819134 52503719 935093966 291904104 925376123 804287978 211983967 244592288 850800567 266353440 115672846 136942127 864735496 378599597 825342270 983403260 40925208 72654596 401705775 549353465 772058929 698323644 187871315 639156817 400004070 967318375 274392755 274267619 823635846 579945662 157651494 365414178 523276402 130813770 677432907 911458327 976656155 181372037 32645986 622970937 234291641 784629157 689769099 497876923 396537384 881381574 12276739 
57
149336403 915813983 837989282 352967707 697547985 308101221 256074304 584393093 210974069 594227871 30232125 737835767 742315580 601113671 354132740 897003690 323085342 425946942 414837289 721627300 271663522 419541116 509467450 254616464 621248650 624206803 694112566 530463583 910887551 949246823 480818139 786022013 285705337 833330536 194193144 146068536 849065512 723791224 44683105 868255470 668845778 554060919 703196819 243697894 365689867 900940146 358353444 371498131 727038960 649505542 724883772 929407644 794087921 155829300 327559732 735289147 296154706 
11
183298896 199876385 40323647 98338053 358777941 219263625 564233067 842160195 583198738 856003806 493980745 
11
530132043 932817098 47069972 253632035 161931152 202318379 55286280 719840033 807068029 420740672 772538360 
44
768087577 989248067 636414397 71366769 400248519 88303637 510662203 80978373 154711283 150136234 127831761 701271398 153545332 845402966 372612646 439325205 895867883 949551500 970889121 584728407 573924959 167794943 470905293 392926281 516661667 634019374 384964473 48508601 217246493 278090240 232455940 800174648 819170883 844220281 974855961 832219850 300808163 324762752 441261317 319197509 549231426 789731524 150320745 425015465 
30
906219169 835324799 710686384 963049618 616162709 527894203 812389756 851583880 80611702 785735436 414152537 712341548 280247860 7576814 67577762 138864585 606393629 189045710 346060916 519872132 277631357 197137337 601893890 401263355 730274562 38435996 865150338 733053607 389114256 9729963 
35
46858238 800177981 31803312 312633661 713690806 914856884 230363554 854373621 804804251 233942552 873167891 235625504 675321323 865044329 220760624 665484977 211334112 617247648 317003622 264922061 878680516 659617109 480800044 99103207 898749431 374323627 45950518 288621877 139807842 601462314 684518482 781860966 260407905 76635622 233425212 
74
828347823 913669998 481749367 484085337 902180875 347359977 449752562 115091920 311470735 929402358 868883982 101135103 718916640 759974513 167792861 566320528 376624703 439689308 337483735 421357070 623143807 423394977 251285644 709168744 426598924 583359907 244755195 956452154 170218702 831465346 169911440 911611176 588758703 696103628 84774363 88019409 748397193 699843931 713498202 32465546 877547126 503975561 295289940 83900449 259174065 96659764 550337634 834197239 29190102 489632964 29007216 860367471 794882192 282894553 55202519 399701005 791037879 642364878 140601341 729509005 156418741 537936187 238323675 664077197 754965342 665832660 648611978 498260878 41305388 484415105 108661521 149625557 517002920 954339743 
71
202862236 80959035 651813902 49423773 436608297 714404765 557326954 867352589 816580983 871511391 936444920 190794619 180386872 491663602 457895549 624199315 894141251 24487430 412036157 535937440 334856429 817294976 999544089 467783480 348149788 258968149 597657369 154310998 75323940 185017380 621959307 939225367 195393330 70683658 66031616 261404824 824847251 753326898 999566774 251805798 432378614 328761261 159580421 354807469 416003232 70433167 730619080 412679398 983225523 505124428 565226968 200101903 938828888 462211683 603934269 273224126 339478535 109896535 692769969 15349402 715049031 423566365 289137066 838749053 427405105 13259031 884486583 284124766 568037321 963777844 921121272 
49
851481674 926760420 233978385 15693791 388747975 93587476 146449540 789724299 941280113 220630543 636815248 234452257 728457271 312267710 527652303 256862761 61555694 445005781 175196818 466585103 899100192 413641613 157193643 945436190 101057096 34272301 483474960 849815205 995942631 716535808 745098599 64181834 519600806 564217801 971053407 881037339 352712461 159090916 509535365 131090503 126247951 601731889 2840793 139174731 328951999 469630704 305601564 568852058 645013401 
28
137675965 752000610 76316809 296898682 22856131 406845156 4619970 535158598 37499191 142624844 774663233 100944657 272520486 194716619 134681786 198638200 64528660 771044147 797321997 811167138 479702667 34357615 396324861 161581407 823012563 714319488 559064840 805618174 
59
99001970 753368747 244414504 117017450 124473566 140263319 953963560 271228063 65078204 249739315 281242181 16265329 582267182 2474875 78617589 991842966 505011109 317636528 324718834 961606636 14098907 211509596 633281966 47324333 149996224 467505521 294899810 654363735 588909215 585217411 887547636 428910864 234375481 135774888 169428027 388172720 534936878 91159665 164926298 500416245 765730678 664617475 401061321 428599548 222035728 635255897 669289311 652682154 398255090 52756653 143399125 842295777 364266208 548547593 828773715 253804255 855102819 556350440 634489860 
96
229095107 88298993 471292552 548890252 964950820 855240033 968060012 697809067 279276173 561688503 515815339 51677360 140627074 386927234 224981261 672794685 438662896 570190002 431555315 634024053 512537921 801780456 329675712 640203159 395805837 7831785 118826852 950361166 604888248 358208410 347886784 549478831 769986533 929145184 285841671 994291061 400350692 644079322 276676307 162600433 797733187 822105465 810632543 377762883 322682941 398652113 36137381 115497544 932501130 30098800 779669385 520554569 267941801 642121969 30345591 90342917 403813752 277973435 223109375 239850140 821627762 234035310 672479894 902460086 118396500 43902046 462388949 913914348 236098322 818565060 287186772 334003959 69839021 414360870 398621412 637223942 701357365 795803749 137618105 230768926 472961913 568859144 72492533 189258572 491068072 334344567 164726047 757823161 633590184 686440165 67431933 208985157 904623180 767021101 560815602 34966675 
70
936222860 136546256 145954422 863534627 61940063 754067564 143181570 41782779 990757729 84804693 392033272 706850881 429032475 760021213 904313929 895728129 502700720 612543717 541572414 46810588 415331359 722309278 605425471 368512320 826570487 851912195 448092277 79866365 51288172 796667439 677079249 627811717 223728869 583418219 102773074 383666378 514905348 888649995 752417348 285037978 735719095 442854219 198677004 165262784 72532012 54276379 244705972 705227773 636954278 134862709 228773887 24525156 516670009 390418270 100168274 278322325 135202139 926896826 53665902 21226094 143690297 965499225 413956059 521177884 827242235 505001700 883107362 945068865 435819455 41593183 
39
519697154 939974026 61706023 905978509 824611109 279491618 153737807 394387983 137596343 800377521 994902775 744166345 526765724 397894108 498061695 733635310 482195144 282909738 667973243 326452614 475624918 981420705 669750218 738145799 298541438 514169922 850328189 997918336 453040082 646906342 410918097 818630283 286592914 963937009 524163009 231955862 7870014 630622285 454120884 
76
388769334 808755581 134562445 45652195 351003484 263128403 317922069 296283924 348736022 889273252 371823839 430622924 530861645 484941321 767332225 48846930 862164204 805799456 322455721 183144510 367796482 143405828 133353960 776178311 826249894 521869756 692819830 625202511 382080163 139072080 391809045 288470486 283849166 779150592 572699278 85424459 13827008 164001771 29313267 921313092 513611024 932244402 113888786 239094630 738097606 639808470 733006139 943408626 124500425 852346545 480131870 967093754 769159476 731722524 775511206 717557836 143470580 711555066 24232638 79532156 725807394 646719586 974696073 876559975 942985173 207714127 731872159 590438693 716277082 932126488 596231273 418666872 38614661 194336827 192417569 539345174 
92
619073337 244869515 470739165 115857818 764553146 531367533 771163188 727264756 581660945 64158994 123733971 66826913 877981532 630325154 959741047 686589968 441391370 442185625 813533382 863173565 850933469 280321318 247579627 540867344 876575071 197562905 610992525 29905148 761918705 131829692 59574957 859645073 637457626 288827480 381592567 132458679 464990483 756349133 602341177 762145617 561842406 689594466 927345488 670554880 148848130 536192583 78169531 561006702 110400706 576920566 51530581 254253036 267998499 247416862 338281038 949758531 655328271 785204247 380925029 407462794 152795005 642749091 199195489 368066081 119684139 639868383 602216406 655591458 658810225 99996231 653527339 164650598 651211868 136784116 544891094 938810036 432530855 678638706 838419229 234520294 125144995 865478262 915064011 399430050 323959053 696705627 798616028 114752829 862516008 71111378 659169033 935899007 
77
527643594 46263322 589651963 607735882 333431925 136504654 769796188 661835428 29921085 28803203 453695569 356411163 438856820 212454626 771626860 635562952 10781944 134530338 934885107 743740381 426051114 542371148 789857204 431404320 194250790 664706026 264410993 7179931 827881970 364159619 156585781 820042168 755824433 425027737 38451467 614536388 811783381 976037458 864710387 439710184 416756161 281521562 299381677 804833034 811586093 793893953 991304596 867120853 951950350 664316443 257796820 998506563 230683197 728641032 467868512 446774247 144043293 269734588 732143190 713271678 652178789 618565196 27766659 768589885 253832899 216316479 912842138 303260460 590568284 410409162 61056323 124331097 607983216 215251478 72983714 271708167 269159083 
34
542173209 542694141 443947795 850185025 7559380 788297585 807621969 97087045 92386500 576305007 255011964 809862406 507086726 288109180 967983759 128037480 756291310 212365962 590591308 564812164 422295732 645970749 443498713 498883377 700551647 690567552 346825322 235845174 508814108 252626799 259938586 612451487 909316292 161717113 
28
3062643 695450669 512188999 931914207 285198785 493528253 193713545 415864866 127823102 156965482 834372498 287387673 390365780 583669133 585099169 842233513 74605419 261937932 797460756 908726511 332003799 85012196 755338379 512667240 256416300 89701622 837451310 452523761 
28
703256598 26854786 660043059 49971096 365879043 420013053 955830200 152496598 982176416 307762019 614719343 832119946 464442536 876485583 739037935 967744339 507081931 230881957 738128727 123621829 852241986 429923832 771767978 67757836 515020833 53904468 50936699 195670028 
34
135957619 89969077 539224847 985175588 401714542 24585162 141876438 309219692 506382343 911349753 566294357 630745642 466309963 385047344 543394204 44224996 471491837 920198241 678764066 705713372 587888217 907416136 986139707 399707584 602672223 72543153 767766342 897909673 576556769 253813082 341936690 841640975 140266567 909698094 
53
561870510 347307614 332702772 145463765 335336801 143098182 160707657 945996651 415025556 984216308 879147076 711291356 477161433 165862030 729685307 449738866 960958480 388918045 654708758 232972230 934160352 337561985 268305902 104079777 203868854 727062389 845461906 275544383 892197412 988359732 293564578 490999094 312723389 114577606 545887164 100426094 114485228 414403631 352116954 253124048 344749508 758367537 237537845 881204545 245774066 327146994 590416180 114354709 227714088 813594234 634723229 420982124 250763764 
19
983283517 4935575 488531123 448488740 25012475 945189955 902261754 528660902 752415644 991673026 918627450 305902896 411492002 873138281 132912483 649107495 147622927 407160046 984056958 
85
308603340 723523203 868424959 636364211 345308939 61520872 26286885 640341533 300024417 10064957 767391207 328690358 809540497 933076364 174962537 185095158 421545514 656608625 718572885 425805210 630785160 405483213 50640386 661609707 163863067 894942099 632387669 55337724 169618599 546495736 466407586 160127716 408514494 539204849 451145569 391945888 317459104 15355425 520694276 172688727 164628731 231118702 788850904 380611863 290366041 475796184 347086395 687567334 99826724 535776623 237439118 297172809 60603027 253211438 438924898 240316200 889884250 412636367 889597266 283879635 339829795 493444840 883007096 690372975 745536052 369204965 1943383 934481994 220574028 768519103 853230629 659259599 678346298 50769424 744544343 354824671 139673605 756515675 439685209 60090797 500610701 936164612 177295547 55225494 9640393 
20
785135675 259353099 863322118 784359192 63649108 437162322 271841667 242945305 318310382 375764030 229984756 202277677 452170177 405926027 285720548 500995551 484497873 572838420 632776502 605587172 
33
441594449 300420329 273662555 885785537 833983802 9276391 491750571 550964841 439166579 719271476 887958097 65801887 32676711 173294376 237598470 84901141 16976024 809536841 896142737 202958887 58600050 378173864 697359159 648675712 987920882 890332229 142843155 840109123 658764251 631074298 841867112 843474233 443927982 
20
748464140 514541296 247727970 431837078 316778171 484950168 860606370 368663160 382905713 882583312 857960520 887484075 230624155 424182730 650894462 88225082 635558125 97270931 149266041 802518970 
31
183735877 424127171 93008204 561016598 533244491 200351237 517821165 981170246 384011923 986464610 701931107 379232902 976781871 904956745 973780439 794447040 264443191 75834622 38844466 428819128 489009981 755738573 138291022 463911437 643602682 919925561 282164377 75996327 855180215 986667702 749973401 
36
155502501 942432420 246738413 996874161 769599389 760152605 890965263 764596202 62529826 483914167 491053046 299522348 416765948 935747541 648971012 409580145 922708239 157580418 420611120 474821917 8273153 835411121 543742141 850353859 161488246 435012804 544255280 9400100 591663333 230274151 73888112 288805643 297396525 618835845 262372119 518890507 
4
897066429 146485558 186858984 284781215 
56
842712622 901979315 865424702 81676636 272582979 920318250 44800256 300625635 734580225 756943067 934166103 375479286 703572549 302146508 373151670 742690267 414956177 679854554 593247494 190152730 96667677 52751133 794874582 709104971 589180523 82684506 526941702 651429153 320136815 128314921 468767756 922365496 351549081 796086988 608624601 966775056 969764978 482827663 759750530 283679847 348991793 707742225 809235330 641918798 592387709 293550709 802303321 361452528 781976708 624935775 92039365 92387201 838771371 223614435 839627266 13164420 
95
106599895 931387765 979924805 883509026 489352265 368256485 160485606 599560026 134710481 770680442 439944242 824547785 791342996 460090561 190577098 883964376 924017646 208236972 728374515 325593562 576266466 220126494 140270186 97421760 656382059 340858721 826004753 725113465 978269747 779497667 842216054 13358076 474823643 809214423 516803811 823510345 953533865 823707996 187890087 176945733 276978508 749035941 437853319 716178243 822068987 664288015 538112630 307113812 627027189 15118027 276941796 502097783 706089631 821099698 449553471 413692200 297256630 331929487 823177738 264694886 363572942 85544065 610205289 968663812 354386523 121380395 161203932 222407394 634347220 342603445 344462335 732660776 132980358 47912487 844397008 863170544 578295464 257138954 897055106 853010858 676949284 762086766 148588231 304356785 277612476 49520657 433805528 415080991 471447446 461451291 262433461 685760246 295311050 435116072 173613305 
21
356883640 200335203 325417010 282294578 251996463 980847506 562129208 828605985 989414024 293714150 221269606 809980891 871716961 543938646 493447409 146169097 50605755 199798838 898491949 749961186 480948268 
20
210298238 590760750 560537027 151508790 256671428 347801562 917354894 987183083 497105850 430217263 728544490 691854 434024572 259161332 360412179 244411432 300953069 209371592 980321828 740564381 
53
778359933 624066798 120815028 678402354 155213652 889790598 902162694 370762894 879811956 293350966 50052783 78092238 758747477 652144315 939683373 372322636 150228427 152995973 495245517 828568569 129879775 199750799 522777730 746803308 451481244 311944011 391101348 746782285 912100401 531502500 157662482 870151727 887109848 665362504 614368271 203087655 229447893 467937417 948334705 440067805 296859201 935551096 834670484 222680203 742816891 235356211 506090603 775035910 798468830 707115237 107968924 321685845 801987452 
47
21754791 268650510 290802731 923993090 894585575 538374622 873752770 268176670 288950668 139966455 564889532 391795110 837393470 51506285 654703269 244782283 239670714 290699877 314034154 870512387 48530580 731779521 937045400 389614693 218004041 473393819 71179164 133906725 289533959 61774987 535461947 296864480 95971636 214328589 300486033 923607144 826292240 819606782 356408227 538233480 845011514 177942898 70100886 692854365 155687131 155433903 811716096 
12
124928875 691882040 289852897 106021847 217389095 725974197 68292341 518456254 255801351 819969660 766159706 849656607 
88
382362175 808192730 622859145 624714195 893592887 80664966 85725970 55021090 761320725 875175140 544822994 142196757 49344124 890908081 608496042 548843112 872090778 822170832 736302142 412626032 49333341 511166408 153708764 880409559 661163695 825064120 185255009 849300714 287900792 986041371 83571730 291168436 523444621 608604358 281949682 178704095 529573190 253745889 867484722 208120782 69881707 660986637 167890445 77791453 779658920 693362706 653749888 697485105 394327790 73608961 261547315 804847756 9309393 401386238 12800342 195172873 462320405 829022597 121457863 259345010 872641098 143436991 354135861 157595199 811955480 488127669 183308064 986950800 148787253 646466162 171800522 847551167 867383921 669527386 753086852 282862854 100260897 753160761 201619921 501538444 150016054 745284279 667234966 51155643 131977979 424675276 949922541 701904107 
25
669585528 595023121 631997281 236507543 943416029 752046754 262297421 737866669 127587992 258134621 81283233 795522255 129286574 97260528 94214681 609142068 136507690 820671356 768692174 729664026 472441976 665867181 40018074 970653581 767628904 
43
582343188 539503572 777992990 668805913 174554543 975210252 926436627 800709263 505981630 971535455 295251842 554006951 96413766 243279580 247954129 855862616 885975450 90365860 773718806 642888803 236922133 229960384 46206376 979410017 34917063 296994513 435680994 885727580 336688282 716800553 554548029 151670857 616038505 973557085 657813126 381106409 347097380 539545489 213627878 71551421 434746401 437411132 506246645 
74
561898253 397430109 163431319 232105343 222607373 133261538 120029375 700114395 977867767 205034390 622537861 677158101 39612040 28311818 457817773 231232879 132527440 62136994 606106488 389348309 565580442 344957778 741022361 549845580 950433420 121072454 989352029 927058217 416193722 441429230 141527103 966103706 564076316 366926014 916974809 505008402 624018333 616196494 445053935 871569764 237085864 544197600 661561735 632901309 525227570 280057963 862004237 723933508 75203379 737870622 259223619 288030387 627290835 125007735 367355504 748067422 218380015 98408437 353371098 369039559 385242819 404490598 828902551 285056651 475692284 653432932 293441429 579399173 431280679 185639349 607788423 413862476 768155729 169273926 
75
996001023 163575563 654453087 631842488 605871242 984802706 167967078 581777700 687379919 37632636 338396137 934248653 621522463 155298882 222899507 855030207 258832713 785683952 449093250 815818926 938730805 257655147 279213414 358968782 332663864 526228317 80647835 323123379 769034981 509928473 898753010 607623390 613293738 280244521 346499656 290195134 997653330 954365119 4057683 820209519 628068419 499307834 135589791 13655490 772219483 466567124 97182814 901914305 431438030 339011549 807861602 203942768 894197944 929573297 557527763 742845804 179327080 530039317 379442939 24726068 781050680 264056262 328649530 276980794 194663386 938812047 196731698 159668253 190860324 64271383 518473355 942977376 306063592 987398905 989162220 
42
493924730 886053506 919931528 669994081 107094336 256237021 839071866 788715435 498482540 170642778 835928016 469755662 57111660 254490618 949116277 291725686 751878807 891660526 553451091 380325697 11301832 229310258 248898898 515697387 732438658 726462339 261397684 170877068 202710796 696519755 347364246 821257004 945703833 923926949 949730742 64277743 7826441 288188764 985407679 841317751 531015208 258387950 
82
87285349 177644926 514124594 847572945 849853671 375590783 415875760 314041705 758835404 294330532 528259367 744002906 807730640 134189211 35014389 224790009 495012372 103721279 437776739 53733086 267110666 30227853 932956614 723109806 305858594 286008630 102713746 569767589 200019394 803972160 107020712 432088839 979516988 294815098 98396781 401530446 897199423 373305780 331324673 407314628 119041911 120375753 524278196 413501902 707511022 194665402 689798274 305627046 854312335 102546357 948602232 995192794 215664357 739229066 892834309 631841631 876913695 835279061 739652196 74041949 37753419 864241046 378496895 591219153 407761775 606792607 148267989 154726469 248315370 5330980 631401404 186965307 95092536 521796546 130003263 249198623 814731488 967902388 849154020 796385687 911715387 686700831 
73
523929138 974466429 734968370 900922793 953406579 424429536 543395976 188684674 3972382 930791231 17463949 526030471 702475138 844965148 552623189 942866860 212428490 304984005 699138965 204620071 937849057 745231315 87766511 926235130 54363841 600997380 397007101 43858221 452051335 131190955 388421343 948091539 82657869 589768638 440128819 205100495 130722690 236350065 241408693 94971841 469053558 687811516 698967221 949640947 329825354 861574397 201768422 721363193 277717014 524016035 43089319 990870098 543902786 730200777 357474114 808252775 346562635 559469293 217875138 5427327 317935861 678815480 96125524 981342969 609816657 264480317 451764683 561036271 778588102 167963564 986342790 905603007 665687739 
56
885636515 289054539 657864780 795076868 340380434 877484651 948640587 907116479 414854908 929406615 3155077 927047565 203756598 31403280 938772482 598348720 559954705 376207271 156083153 592760314 127977128 127319760 382388219 463372588 855840771 134696186 812255010 53152570 56487979 359966651 816005813 370053904 25385720 455081204 87798665 200641927 675678499 738623120 263307057 572805653 360550795 40884747 441804987 271139889 584925732 167870524 61937414 556299544 47800093 613733820 910365373 716179433 992729096 398108921 353922530 105010871 
62
271351669 905770992 303561061 220506427 281475236 879367740 824762021 22116441 202440492 873954929 709369681 595021350 825541100 6461030 94784197 376605684 67507184 185108463 201738224 304455315 895039876 848373924 463580873 104712281 738733297 827380258 703646116 734264858 181948190 623933291 899481298 188716220 217065570 104299673 319118493 612492693 398324832 258530844 534913330 262560717 511376727 789119654 845108427 58807372 323721369 200419152 62658434 991210633 154109533 649861620 695591394 960911778 703234374 594226077 394321751 756692243 766787323 935508026 563158818 896672968 494461751 95629971 
61
287805351 601495518 401194251 869620030 13330853 148275746 319441 246946821 670633839 348393251 494865568 659889255 477890448 488259309 883659869 872374471 568820229 517310748 730053221 184007223 70152058 10408152 559609183 867164132 822813436 285969328 819180405 341126814 213976992 351013925 937578933 741116503 31517257 954077429 113099901 826470570 667141261 156261161 434675271 21755335 270361750 785339054 887867076 538795779 157831318 361813753 307989185 291955882 140849632 31178140 628980871 496373624 113706070 258780455 842910620 318814680 426843182 970940443 350771106 30261505 487503 
17
102313038 597689065 471494143 114884645 28047725 991981722 844168375 735429874 271706864 7143934 905859279 831247822 692072937 172534015 137162225 751221998 393850830 
21
87062839 162400157 921832941 387102177 210596673 147588414 343788664 192328237 586323719 191584482 344206075 83908525 58207929 642192304 165994457 653017399 146749924 819956618 376605868 84084925 614868970 
25
317107107 138423324 547817955 935465104 119377425 852657640 90359839 292767107 327473496 258093562 453439990 658092433 495669343 571169075 963458569 364907657 856939207 118389369 603082106 633766253 205506142 781622594 581934459 333864099 700467207 
99
409705832 616466301 247037735 618116349 835317975 639307454 317667195 250296107 19849367 424327484 998301494 574035720 726404805 986996138 330431620 949849946 227989244 279767315 30936985 812542767 921025627 217829752 67473535 89297064 68685978 21445347 373429109 164752751 519971368 133602719 390535776 804081237 977078724 124839475 99958943 569145142 483849826 848671331 902237586 35353928 147385722 456626462 733208678 103846076 44049530 342516885 949708752 33323643 930615748 288355649 124014943 472789580 536842963 51500925 281393009 714922772 656689321 544495076 828131125 431030662 131250688 93249369 391694703 611494398 506972444 390793381 148017707 326483345 836607539 345742102 264805625 456573283 547842744 937710384 586764736 531477188 946074983 645003316 647991830 402016682 878517422 83022671 561067054 902824875 608528737 948272960 277724517 606390148 487797207 592307187 886671759 611652355 408975203 783872660 21772794 272305230 270869197 268303127 695360823 
65
781748509 198393273 429898577 401177739 631212512 277779389 497802872 563982369 118055222 114862898 91334954 948723434 729880182 277329880 237468836 371999309 343677750 107259312 311042310 43376339 171637761 785206121 751624126 926313276 492566063 558719538 309454791 61018785 558197179 263461129 231466539 195366985 775039497 48872373 242414939 105004365 698083730 498343137 874821892 558886931 222068202 178441076 460566303 956247670 709019853 117158274 282383409 634179316 948202993 629397042 340646686 763747819 30247552 941584734 453042442 878278382 718826808 191759320 2375653 944228724 832865243 980558312 436384063 495966800 395130728 
12
567925240 465400220 335640443 132148699 141243903 410444728 913492108 712275215 478577355 777071617 904717832 525887924 
28
5697086 867701271 515815505 908345573 148681514 108254382 865126627 724531491 626391428 373311364 266066087 926294499 51761635 369745659 159680446 208894894 527699610 618972267 510008260 169349124 846771505 443043609 817140447 291979263 539451709 532046865 552734526 503234131 
35
290003813 188298601 743552997 856954107 86339050 246441031 426554069 114034500 475435862 44257057 577025631 786941464 510581937 604371416 258574755 12717149 104745485 1669884 352481849 400487397 905246804 967741881 482958107 280566582 591972387 490815532 251832755 499892459 809668360 429762060 828460338 511693905 155578393 349494285 701170025 
97
261403364 453160332 508627870 731737318 184626491 24793987 392954029 985135090 376557601 436208656 37591104 154904068 728128570 422990148 589435175 65357934 316120968 573856866 299169020 574835138 488542846 285524962 678727061 85373567 56203027 401041290 805767271 398319097 200141881 467642960 161895171 697330356 298102574 928743471 228247134 228249198 272858497 500998631 421314505 691649594 996301934 51547339 432068141 108728489 433005819 248346441 970641657 212973302 125175707 68739030 92831901 477873252 658607143 813152616 97027453 940148125 900270272 959814640 3331637 66221103 175882845 394228049 273188606 519993868 98442824 589410911 52551314 120318006 256950929 423283832 993233772 402201934 420879090 231542176 164140378 115202142 875006837 783961230 906680528 416503597 355962190 15321690 968818168 13877695 512821934 518215585 522243309 621228264 58816729 103092291 398303026 391334591 587765567 451814969 61799318 537048540 242146639 
92
288128764 80715255 857190483 572879326 263602839 437130368 84586283 521592910 597709142 352752966 898929930 187591557 46680714 680490585 587199567 195365281 11327032 347799489 635099789 423060350 228183371 591417763 524235947 789048371 934731805 587531150 56455823 88538999 981259977 776295122 240569945 753407848 497386238 728441137 813023703 221585507 19678772 60278305 661817676 266042101 655488905 192453451 551876545 135129452 768786789 153129540 118867155 355550802 774853567 374545545 909160499 794298860 445104884 100015379 218822888 562299907 514770499 905825152 393200132 150617026 206320305 665178919 281071492 186153559 790484921 223713001 196415977 46175077 591589746 283498180 773578841 51427771 568418172 508384706 575052931 69749161 584366637 917115149 162846139 16251294 772323763 367145349 532171683 291766374 700581884 617072857 608621409 113761724 627037969 948935182 9364714 413038296 
42
937452339 980653006 194284873 236884376 251372795 756757516 191524023 355841910 181243650 483548872 566291895 258774395 53290970 300329359 947113248 4644305 563480178 284666400 666379679 248900819 815940682 142273754 355186895 166798624 425104959 744432250 664674211 356272320 945281844 929323271 837020187 612338848 239841652 162071397 721939042 850256837 255794233 627838407 555478620 237343022 179715277 627740789 
73
665031327 644369154 46550844 903532155 760677215 330243121 276471974 156747982 191530646 28279044 64279886 299706851 169350397 142866534 30563347 410933327 971884908 30753752 240112640 534051403 107668050 40758270 621199632 454817340 566613395 592821689 142610949 738867734 278678444 524900875 909104559 767736453 364422934 460997606 527796086 109422423 294125009 914385317 589020323 445176945 568202283 222398945 465679624 622265504 935508681 342056868 551090610 886411214 674083380 425246435 773178992 799543488 760645358 786563686 620266820 935173829 880991608 612009292 54642814 433310742 717857881 216254054 121790547 750877370 467414651 90395663 274213346 861283130 653156672 196625816 307316091 90955856 962826075 
13
555775510 653498583 126184117 230419392 748092326 347761304 201121538 622043359 468593358 431201688 317772435 165845075 375523940 
13
443881978 352945575 999878286 914633727 372112002 89532545 488287529 390918124 643940511 153015420 851147961 659636937 327686312 
39
902376152 14986896 956476646 974454340 29781020 6385086 777420979 405924183 587086958 625656261 759150471 817221430 518856644 395421774 654484777 127430720 524317340 983439566 208041888 870760843 790113220 70077763 799810538 8913437 131425411 483840908 966292340 900497247 760586175 90274072 152638487 600202345 589476134 321459949 710320006 368755172 435494608 666824814 755428363 
6
785957803 86802316 921547197 221318051 133155908 81914356 
8
50288459 903462690 657702348 404546335 348302254 30743193 152209106 765734705 
71
630607874 278246379 721005448 182079834 453724091 205679736 205681139 548332805 543922029 871698438 129282668 307329264 847930432 967343589 286363695 163576991 785553081 596748271 246868226 560075229 487874540 62803042 553033971 69269083 835845312 32410253 472219925 330382719 203239465 660136019 531057715 714131265 749824008 867834157 204412942 424501736 347140462 754908968 952125754 312749293 61804604 353372645 567463481 335309733 254221275 927048262 875796204 342223252 922135766 730379720 858785231 282854857 401918084 587389675 149586931 273355137 657099310 163316778 628012725 93554860 23472412 466899701 897390827 159088008 659117890 243814668 272586174 600106944 145886453 164903497 966169485 
23
220344177 280356114 625026596 303012128 254377092 953936656 13117195 557799192 223825698 991181249 546458223 202567650 367431464 493504844 784705322 486021019 424719040 48575484 209913883 773869267 969898173 696208963 652222434 
80
879259082 958155614 102614922 407941088 215548844 235362010 186826275 266287806 645872929 964210323 960816474 671762823 76216694 415381240 458052203 233126399 476503387 349649043 95396372 297719356 299045104 810539357 864431453 593776770 680305408 188262832 126378519 264985900 198944228 888471238 258853063 475626702 624804251 36508368 282723327 327005992 162331066 823664124 103096057 184138895 369847305 231407411 377427767 666399307 418408313 763307739 110486624 942879560 201783834 996118044 231216279 783487206 130207870 90879904 263370874 376255223 523973500 498287556 148268249 377294540 786563433 801746345 180990312 207323117 533217957 558613141 928894656 771972325 65804677 555286957 374487859 758408637 137547578 856817750 17894478 59438174 438140729 313615904 131923282 657223656 
49
56694390 910364378 698064626 556121460 975664128 909473899 543579202 666122581 69185167 562935672 602352036 444585758 814695067 467648343 280051690 524516283 692992841 611936973 147163362 288319786 786264912 618295269 983648261 217100201 49316931 802228474 660889151 81108988 173508347 193724600 409515073 148146668 154365992 84650703 4893168 5784255 773324995 141512042 373654911 681986806 127453251 902300552 537802321 38679374 229130692 805857722 863409518 856199388 553384747 
