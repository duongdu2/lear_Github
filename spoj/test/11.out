2910159050
2933852632
2794468450
2281362739
2512423487
2934993149
2734419125
2692286831
2840194255
2982336386
2912432228
2439797875
2907413162
2927301893
2901325819
2974241816
2885198453
2848309614
2965848617
2727485076
2682874231
2905750955
2891513536
2933534520
2959013501
2803722970
2432816985
2774395848
2630673757
2954302558
2875054122
1368706628
2870706137
2926858364
2868753479
2884859805
2823569174
2784645634
2435785973
2922914712
2734740966
2928177354
2933323502
2982816573
2884254698
2942612170
2958555857
2880142165
2887630437
2862596805
2742009376
2394806958
2755862880
2962376356
2885529416
2595281557
2702341343
2729942792
2974670796
2864926964
2873048098
2870219664
2023634268
2765659974
2924186246
1928823051
2326899743
2885638828
2915016078
2921144841
2869676767
